experimental/hip/registration/driver_module.c

```diff
@@ -49,6 +58,146 @@ static iree_status_t iree_hal_hip_driver_factory_enumerate(
   return iree_ok_status();
 }
 
+// Option key constants.
+static const iree_string_view_t key_hip_dylib_path =
+    iree_string_view_literal("hip_dylib_path");
+static const iree_string_view_t key_hip_use_streams =
+    iree_string_view_literal("hip_use_streams");
+static const iree_string_view_t key_hip_async_allocations =
+    iree_string_view_literal("hip_async_allocations");
+static const iree_string_view_t key_hip_tracing =
+    iree_string_view_literal("hip_tracing");
+static const iree_string_view_t key_hip_default_index =
+    iree_string_view_literal("hip_default_index");
+
+// Parses flags and environment variables into a string pair builder.
+static iree_status_t iree_hal_hip_driver_parse_flags(
+    iree_string_pair_builder_t* builder) {
+  const iree_flag_string_list_t dylib_path_list = FLAG_hip_dylib_path_list();
+
+  // hip_dylib_path
+  for (iree_host_size_t i = 0; i < dylib_path_list.count; ++i) {
+    IREE_RETURN_IF_ERROR(iree_string_pair_builder_add(
+        builder,
+        iree_make_string_pair(key_hip_dylib_path, dylib_path_list.values[i])));
+  }
+
+  // bool and int flags
+  IREE_RETURN_IF_ERROR(iree_string_pair_builder_add_int(
```

Comment:
yeah, as expected I absolutely hate this - fine here to unblock you, but we'll rework this before we put it in other drivers - this should be something we can make the flags mechanism do for us with a table instead of needing this code (and type conversions/etc) and we don't want to be doing the conversions twice (with different quality of error messages, different validation, different type support, etc)

Reply:
Yeah, I didn't particularly enjoy writing it. It seems like the knowledge of what we want is encapsulated in the flag system. If we can expose a pattern for doing this kind of thing, it would help us not repeat ourselves with such verbosity.